pkg/controller/grouppermission/grouppermission_controller_test.go

```diff
@@ -0,0 +1,84 @@
+package grouppermission
+
+import (
+	"context"
+	"testing"
+
+	operatorconfig "github.com/openshift/rbac-permissions-operator/config"
+	routev1 "github.com/openshift/rbac-permissions-operator/pkg/apis"
+	managedv1alpha1 "github.com/openshift/rbac-permissions-operator/pkg/apis/managed/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	client "sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func reset(ctx context.Context, r *ReconcileGroupPermission) {
+	r.client.Delete(ctx, makeGroupPermission())
+}
```

Comment:
I'm not saying this is wrong, but the notion of making something only to use to delete something feels a little awkward to me.

Reply:
yes I agree. I just made it under the assumption that it will be used in more than 1 test function. I only have 1 right now

Reply:
That's fine, I was concerned you were using the function to create the same thing multiple times, something that maybe should be a constant then. But if this function will be reused and fields will be different then thumb up. Also should be easily fixed if that doesn't hold true.

Reply:
This fn is unnecessary

Reply:
removed func